From d02b5e8f6081bf4470c1482047dd90a957f814f4 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Stefan=20D=C3=B6singer?= <stefandoesinger@gmx.at>
Date: Sun, 15 May 2016 16:51:06 +0100
Subject: wined3d: Assign a read buffer when discarding a new texture.

Otherwise we'll try to call glDeleteBuffers(1, NULL). Alternatively I
could check for NULL in wined3d_texture_changed, but I prefer to keep
buffer and map_buffer consistent instead.
---
 dlls/wined3d/texture.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/dlls/wined3d/texture.c b/dlls/wined3d/texture.c
index 0f0bdf2..05d7565 100644
--- a/dlls/wined3d/texture.c
+++ b/dlls/wined3d/texture.c
@@ -1644,6 +1644,8 @@ void *wined3d_texture_map_internal(struct wined3d_texture *texture, unsigned int
             case WINED3D_LOCATION_BUFFER:
                 sub_resource->map_buffer = wined3d_device_get_bo(device, sub_resource->size,
                         GL_STREAM_DRAW, GL_PIXEL_UNPACK_BUFFER, context);
+                if (!sub_resource->buffer)
+                    sub_resource->buffer = sub_resource->map_buffer;
                 ret = !!sub_resource->map_buffer;
                 break;
 
-- 
2.8.0

